src/System.Private.CoreLib/shared/System/Runtime/InteropServices/CollectionsMarshal.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+
+namespace System.Runtime.InteropServices
+{
+    /// <summary>
+    /// An unsafe class that provides a set of methods to access the underlying data representations of collections.
+    /// </summary>
+    public static class CollectionsMarshal
+    {
+        /// <summary>
+        /// Get a <see cref="Span{T}"/> view over a <see cref="List{T}"/>'s data.
+        /// Items should not be added or removed from the <see cref="List{T}"/> while the <see cref="Span{T}"/> is in use.
+        /// </summary>
+        public static Span<T> AsSpan<T>(List<T> list)
```

Comment:
We need null argument validation here, don't we?

Reply:
When writing the tests; I realised I'd prefer this arrangement 
```csharp
public static Span<T> AsSpan<T>(List<T>? list)
    => new Span<T>(list?._items, 0, list?._size ?? 0);
```
Not sure how people would feel about that?

Reply:
is that better than `list is null ? Span<T>.Empty : new Span<T>(list._items, 0, list._size)`?

Reply:
Seems like the latter would have less null checks/etc

Reply:
It's easier (at least for me) to reason about the code if written how @tannergooding suggested. I also would use `default` instead of `Span<T>.Empty`

```C#
public static Span<T> AsSpan<T>(List<T>? list)
    => list is null ? default : new Span<T>(list._items, 0, list._size);
```

Reply:
I believe we prefer `Span<T>.Empty` over `default` based on the existing usages

Reply:
It's easier to see call sites of `Span<T>.Empty` than usage of `default`, **but either works**.

We have `default` being used in similar domain here:
https://github.com/dotnet/coreclr/blob/163b30c625c4a3b080743af2fb60502a164740c7/src/System.Private.CoreLib/shared/System/ArraySegment.cs#L170

There are only a few hits of `Span<T>.Empty`:
https://source.dot.net/#System.Private.CoreLib/shared/System/Span.cs,a02fdd2de3235cda,references

Although, a lot more for `ReadOnlySpan<T>.Empty`:
https://source.dot.net/#System.Private.CoreLib/shared/System/ReadOnlySpan.cs,47744f8b41c9226f,references